src/HttpClient.ts

```diff
@@ -421,7 +421,12 @@ export class HttpClient extends EventEmitter {
       // handle digest auth
       if (response.statusCode === 401 && response.headers['www-authenticate'] &&
           !requestOptions.headers.authorization && args.digestAuth) {
-        const authenticate = response.headers['www-authenticate'];
+        let authenticate = response.headers['www-authenticate'];
```

Comment:
headers should be string?

Reply:
It should but it can be a string[]. This is caused because servers are allowed to send multiple www-authenticate headers according to the standard, similar to set-cookie (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#syntax)